### PR TITLE
Use SecureAbove22Preferences in Fenix for encryption key

### PR DESCRIPTION
**Pull request by ekager**
---

Building on AC changes that @grigoryk and I are working on for
- https://github.com/mozilla-mobile/android-components/issues/4741
- https://github.com/mozilla-mobile/android-components/issues/4758

Needs https://github.com/mozilla-mobile/android-components/pull/4958 to land and be in snapshot first

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by grigoryk** [comment]
A-C PR: https://github.com/mozilla-mobile/android-components/pull/4958

File: app/src/main/java/org/mozilla/fenix/components/Core.kt [hunk 1]
```diff
@@ -165,10 +167,23 @@ class Core(private val context: Context) {
...
+     * Shared Preferences that encrypt/decrypt using Android KeyStore and lib-dataprotect for 23+
+     * otherwise simply stored
+     */
+    val secureAbove22Preferences by lazy {
+        SecureAbove22Preferences(context)
+    }
+
+    private val passwordsEncryptionKey =
+        secureAbove22Preferences.getString(PASSWORDS_KEY) ?: generateEncryptionKey(KEY_STRENGTH).also {
+            secureAbove22Preferences.putString(PASSWORDS_KEY, it)
```
Comment (grigoryk) [inline app/src/main/java/org/mozilla/fenix/components/Core.kt]: I think we should keep track of cases where we generated a key, and then it disappeared for some reason. At least to see if this happens at all in the wild.

You can store a flag in regular preferences when you generated a key, and see if that flag is set when you see a `null` key.

We can then keep track of these events in sentry,
```
class PasswordStorageKeyRegeneratedException : Exception()
crashReporter.submitCaughtException(PasswordStorageKeyRegeneratedException())
```

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/components/Core.kt]: Do you feel strongly about the exception thing? I just made it a Sentry "capture"

**Comment by ekager** [comment]
We're gonna change this API so don't land this yet
see https://github.com/mozilla-mobile/android-components/issues/5004

File: app/src/main/java/org/mozilla/fenix/components/Core.kt (hunk 1 above)
Comment (grigoryk) [reply app/src/main/java/org/mozilla/fenix/components/Core.kt]: That should be fine, as long as we can keep track of it in sentry after.

**Comment by grigoryk** [comment]
Unrelated directly to this, but I was surprised to see websites for which I have logins logged as I scroll around the passwords list:
```
 D  Loaded icon (source = GENERATOR): http://vk.com
                         D  Loaded icon (source = GENERATOR): http://www.delicaclub.com
                         D  Loaded icon (source = GENERATOR): http://www.delicaclub.com
                         D  Loaded icon (source = GENERATOR): http://www.delicaclub.com
```

**Comment by ekager** [comment]
@grigoryk you're seeing this debug only log from the icons library 🤷‍♀️
https://github.com/mozilla-mobile/android-components/blob/master/components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt#L95-L102

**Pull request merged**